regression/regr.py

```diff
@@ -0,0 +1,119 @@
+"""
+    References
+    ------------
+    https://en.wikipedia.org/wiki/Polynomial_regression
+"""
+
+import numpy as np
+from scipy import linalg
+import matplotlib.pyplot as plt
+from matplotlib import style
+style.use('seaborn')
+
+# from matrix import matrix_operation as mat_op
+
+
+class Regression:
+    """
+    Class that contains functions for linear and polynomial regression
+    Both regression cases are offered with:
+        1. traditional (least squares)
+        2. gradient descent
+    """
+    def __init__(self, x, y):
+        self.x = np.sort(np.asarray(x))
+        self.y = y
+        self.beta = None
+        self.coeffs = None
+        self.poly_eqn = None
+
+    def the_algorithm_ls_reg(self, order=1):
+        """
+        :param order: nth order polynomial
+        :type order: int
+        :return: ndarray
+
+
+        _____________________________________________
+        Still under development
```

Comment:
If the algorithm is under progress, please mention that in the commit title by prefixing  **[WIP]**  to the commit message or apply the corresponding tag.

Reply:
@AnupKumarPanwar  made the change, thanks for the tip.  This will probably stay in development as it depends on another PR #925.